pkg/template/evaluation_ctx.go

```diff
@@ -88,6 +89,21 @@ func (e *EvaluationCtx) TplSetNode(thread *starlark.Thread, _ *starlark.Builtin,
 	return starlark.None, nil
 }
 
+// args(nodeTag, value Value)
+func (e *EvaluationCtx) TplSetNodeKey(thread *starlark.Thread, _ *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	if args.Len() > 1 {
+		if _, noop := args.Index(1).(*core.StarlarkNoop); !noop {
+			node := e.parentNodes[len(e.parentNodes)-1]
+			if item, ok := node.(*yamlmeta.MapItem); ok {
```

Comment:
Not super stoked that we had to cast back to `yamlmeta.MapItem` in the evaluation context in order to set the key. But it seemed like adding a `SetKey` method to `EvaluationNode` was not also a good idea since key can only be set for a `MapItem` and not for other yaml elements. Hence leaving it the way it is for now.

Reply:
we should probably panic here with message "Expected node '%T' to be mapitem" as this is an internal inconsistency error